File: app/src/main/java/com/moudle/app/view/CornerDrawable.java
package com.moudle.app.view;

import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.graphics.PixelFormat;
import android.graphics.RectF;
import android.graphics.Shader;
import android.graphics.drawable.Drawable;

import androidx.annotation.IntRange;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * 自定义圆角图片
 */
public class CornerDrawable extends Drawable {

    private final Paint mPaint;
//    private int mWidth;
    private final Bitmap mBitmap;
    private RectF rectF;

    public CornerDrawable(Bitmap bitmap) {
        this.mBitmap = bitmap;

        BitmapShader shader = new BitmapShader(mBitmap, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP);
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
        mPaint.setShader(shader);
    }

    @Override
    public void draw(@NonNull Canvas canvas) {
        canvas.drawRoundRect(rectF, 30, 30, mPaint);
    }

    @Override
    public void setBounds(int left, int top, int right, int bottom) {
        rectF = new RectF(left, top, right, bottom);
    }

    @Override
    public void setAlpha(@IntRange(from = 0, to = 255) int i) {
        mPaint.setAlpha(i);
    }

    @Override
    public void setColorFilter(@Nullable ColorFilter colorFilter) {
        mPaint.setColorFilter(colorFilter);
    }

    @Override
    public int getOpacity() {
        return PixelFormat.TRANSLUCENT;
    }

    /**
     * 记得重写以下方法
     */
    @Override
    public int getIntrinsicWidth() {
        return mBitmap.getWidth();
    }

    @Override
    public int getIntrinsicHeight() {
        return mBitmap.getHeight();
    }
}

File: app/src/main/java/com/moudle/app/view/CornerDrawable.kt
package com.moudle.app.view

import android.graphics.*
import android.graphics.drawable.Drawable

import androidx.annotation.IntRange


/**
 * @作者: Simplation
 * @日期: 2021/6/29 10:13
 * @描述:
 * @更新:
 */
class CornerDrawable(bitmap: Bitmap) : Drawable() {
    private val mPaint: Paint

    private val mBitmap: Bitmap = bitmap
    private var rectF: RectF? = null

    init {
        val shader = BitmapShader(mBitmap, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP)
        mPaint = Paint()
        mPaint.isAntiAlias = true
        mPaint.shader = shader
    }

    override fun draw(canvas: Canvas) {
        canvas.drawRoundRect(rectF!!, 30f, 30f, mPaint)
    }

    override fun setBounds(left: Int, top: Int, right: Int, bottom: Int) {
        rectF = RectF(left.toFloat(), top.toFloat(), right.toFloat(), bottom.toFloat())
    }

    override fun setAlpha(@IntRange(from = 0, to = 255) i: Int) {
        mPaint.alpha = i
    }

    override fun setColorFilter(colorFilter: ColorFilter?) {
        mPaint.colorFilter = colorFilter
    }

    override fun getOpacity(): Int {
        return PixelFormat.TRANSLUCENT
    }

    override fun getIntrinsicWidth(): Int {
        return mBitmap.width
    }

    override fun getIntrinsicHeight(): Int {
        return mBitmap.height
    }
}